packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -160,13 +160,16 @@ class MediaQueryData {
     this.displayFeatures = const <ui.DisplayFeature>[],
   });
 
-  /// Creates data for a media query based on the given window.
+  /// Deprecated. Use [MediaQueryData.fromView] instead.
   ///
-  /// If you use this, you should ensure that you also register for
-  /// notifications so that you can update your [MediaQueryData] when the
-  /// window's metrics change. For example, see
-  /// [WidgetsBindingObserver.didChangeMetrics] or
-  /// [dart:ui.PlatformDispatcher.onMetricsChanged].
+  /// This constructor was operating on a single window assumption. In
+  /// preparation for Flutter's upcoming multi-window support, it has been
```

Comment:
Here and in https://github.com/flutter/engine/pull/39302, do you feel it is worth referencing a source of truth on the `upcoming multi-window support`?

Or another thought, will this doc comment persist after multi-window support is in place and still running out the deprecation period? If so, should the tense be different so it make sense when multi-window is available?

Reply:
There isn't much we can reference here.

From todays perspective I think the tense is right here. And even if this is still around when multi window support is available, it will still be true that this was deprecated in anticipation of the (then) upcoming multi window support.

I doubt that this will confuse anybody, but if you have a better wording suggestion I am open to change it :)

Reply:
SGTM